org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/IHiddenRegionPart.java

```diff
@@ -40,4 +40,14 @@
 	 *         current {@link IHiddenRegionPart} is the first element inside its {@link IHiddenRegion}.
 	 */
 	IHiddenRegionPart getPreviousHiddenPart();
+
+	/**
+	 * @see IHiddenRegionPartAssociator for custom behaviour.
+	 * 
+	 * @since 2.14
+	 * 
+	 * @return A hint whether the whitespace or comment belongs to the previous or next semantic element or neither of
+	 *         them.
+	 */
+	HiddenRegionPartAssociation getAssociation();
```

Comment:
Wouldn't it be helpful to get access to the semantic region itself?

Reply:
no, because `HiddenRegionPartAssociation.CONTAINER` doesn't correspond to one single semantic region. For other cases, `hiddenRegionPart.getHiddenRegion().getNextSemanticRegion()` or `hiddenRegionPart.getHiddenRegion().getPreviousSemanticRegion()` can be used.